### ISSUE TITLE
Social card links can be broken in multiversioned documentations

### ISSUE THREAD

**Issue by illia-v** [wpilibsuite/sphinxext-opengraph#103]
urllib3 has recently started relying on sphinxext-opengraph. In https://github.com/urllib3/urllib3/pull/3064 I tried setting up social cards, but ended up facing problems related to canonical URLs in Read the Docs.

urllib3 has docs for multiple versions of the package and for every new pull request.

sphinxext-opengraph seems to always start URLs of the cards with https://urllib3.readthedocs.io/en/stable/ regardless of the docs version (for example, at https://urllib3--3064.org.readthedocs.build/en/3064/.)

This leads to a problem that a score card linked in `og:image` is not available in any docs version until it is deployed to "stable".

But even after it is deployed there, there is a high chance that `og:image` in other versions will lead to a nonexistent image because of the hash in the filename affected by multiple details of a page.

**Comment by illia-v** [comment]
Relying on `READTHEDOCS_CANONICAL_URL` like in https://github.com/readthedocs/readthedocs.org/pull/10224 looks to be a solution

**Comment by Daltz333** [comment]
Yep. I'm pretty sure that's what we did over in frc-docs. Development motivation has been slow as of late, but we're always open to reviewing contributions.

**Comment by AA-Turner** [comment]
Fixed in 98c36cb749af0b4b0112e57baae878f022714b1b and #134.

A
